### PR TITLE
[mqtt.homeassistant] Create trigger channels for stateless sensors

### PR DESCRIPTION
**Pull request by bodiroga**
This PR creates trigger channels (and not state channels) for HA sensor components that send stateless information (action and button press events, mainly). The implementation is not device specific and it should be generic enough to be accepted. We can also add more objectIDs if they are used to send this type of information, but, in theory, this list should be short.

What do you think? @jochen314? 

Closes: #6878

Signed-off-by: Aitor Iturrioz <riturrioz@gmail.com>

### CODE REVIEW

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/ComponentSensor.java [hunk 1]
```diff
@@ -27,6 +30,7 @@
 @NonNullByDefault
 public class ComponentSensor extends AbstractComponent<ComponentSensor.ChannelConfiguration> {
     public static final String sensorChannelID = "sensor"; // Randomly chosen channel "ID"
+    private static final List<String> triggerObjectIDs = Arrays.asList(new String[] { "click", "action" });
```
Comment (jochen314) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/ComponentSensor.java:33]: Why do you explicitly create the `String[]`? The `asList` method has varags paramters

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/ComponentSensor.java [hunk 2]
```diff
@@ -61,8 +65,12 @@ public ComponentSensor(CFactory.ComponentConfiguration componentConfiguration) {
             value = new TextValue();
         }
 
+        String objectId = componentConfiguration.getHaID().objectID;
+
+        boolean trigger = triggerObjectIDs.contains(objectId);
```
Comment (jochen314) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/ComponentSensor.java:70]: This is the part I am not happy with.
The check against 2 explicit values is not future save. I see many people wanting to have different values in the list also to make their device work.

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/ComponentSensor.java (hunk 1 above)
Comment (bodiroga) [reply bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/ComponentSensor.java:33]: Sorry, I wasn't sure about what type of arguments the ```asList``` function would accept, so I looked up an example on the Internet. I will change it, thanks :+1:

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/ComponentSensor.java (hunk 2 above)
Comment (bodiroga) [reply bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/ComponentSensor.java:70]: Oh, sorry, I think that I finally understood your concern regarding the 'action' and 'click' objectIDs topic :facepalm:  I was thinking all the time that those were universal HA objects used to send event information, but know I see that they are Zigbee2Mqtt specific. I'm so sorry for the confusion :innocent:

Ok, would you accept changing the logic to detect if the icon name starts with "mdi:gesture"? That is one of your proposals in #6878 and seems to fit fine with Zigbee2Mqtt, so I would be happy to implement it :+1:

**Comment by bodiroga** [comment]
Review comments addressed!

**Comment by cpmeister** [comment]
@jochen314 Does this PR look good to you?

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/ComponentSensor.java (hunk 2 above)
Comment (fwolter) [reply bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/ComponentSensor.java]: @jochen314 Can you comment on this?

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/ComponentSensor.java [hunk 3]
```diff
@@ -27,6 +30,7 @@
 @NonNullByDefault
 public class ComponentSensor extends AbstractComponent<ComponentSensor.ChannelConfiguration> {
     public static final String sensorChannelID = "sensor"; // Randomly chosen channel "ID"
+    private static final List<String> triggerIcons = Arrays.asList("mdi:toggle", "mdi:gesture");
```
Comment (jochen314) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/ComponentSensor.java:33]: How about:
```
Pattern triggerIcons = Pattern.compile("^mdi:(toggle|gesture).*$");
```
and then
```
boolean trigger = triggerIcons.matcher(icon).matches();
```

Comment (bodiroga) [reply bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/ComponentSensor.java]: Thanks @jochen314 for your feedback! Changes done! :wink:

**Review by jochen314 — APPROVED** [review]
LGTM

**Review by Hilbrand — APPROVED** [review]

**Pull request merged**